happy thoughts project

lunek1 opened the pull request:
## Netlify link
https://celebrated-shortbread-0673ce.netlify.app/

## Collaborators
Individual project

lunek1 committed:
Created components and connecting them

lunek1 committed:
Changed header <h1> font

lunek1 committed:
Added css to thoughts containers

lunek1 committed:
Fixed the like function

lunek1 committed:
Added back to top-button

lunek1 committed:
Added timestamp

lunek1 committed:
Text input with character count done

lunek1 committed:
Fixed characters remaining

lunek1 committed:
Responsive design added

lunek1 committed:
Created folders for components and sorted

lunek1 committed:
Delete src/assets directory

lunek1 committed:
Delete instructions.md

lunek1 committed:
Delete .eslintrc.cjs

lunek1 committed:
Delete public directory

lunek1 committed:
Delete .gitignore

lunek1 committed:
Delete pull_request_template.md

lunek1 committed:
Comments added

lunek1 committed:
Comments added ThoughtInput.jsx

lunek1 committed:
Comments added LikeButton.jsx

lunek1 committed:
Update README.md

lunek1 committed:
Update README.md

lunek1 committed:
Delete package.json

lunek1 committed:
Delete vite.config.js

lunek1 committed:
Add files via upload

lunek1 committed:
Add files via upload

lunek1 committed:
Changed submit font-size slightly smaller

lunek1 committed:
Removed className header

lunek1 committed:
Updated h1 + responsiveness

lunek1 committed:
Added pointer cursor when hover on submit

lunek1 committed:
Update README.md

El1an3 commented on src/components/List/ThoughtList.jsx:
```diff
@@ -0,0 +1,87 @@
+    };
+  }, []);
+
+  const calculateTimeDifference = (timestamp) => {
```
Would have been nice to have days, month etc. too as mentioned in your readme.

El1an3 commented on src/components/List/ThoughtList.jsx:
```diff
@@ -0,0 +1,87 @@
+    }
+  };
+
+  /*Function for the scroll to the top-button*/
```
I like that, didn't think of it myself. 👍

El1an3 commented on README.md:
```diff
@@ -1,35 +1,21 @@
-### View it live
-
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+🚨 There seem to be some kind of css issue when I try the netlify link on my phone. On my computers it looks fine. The arrow svg inside my scroll to the top button isn't centered within the button. The submit button font is white instead of black. Let me know if you also get this or if there is any other issue. Will look into it!
```
I get the same on my phone (white font and arrow not centered).

El1an3 commented on src/components/Input/ThoughtInput.jsx:
```diff
@@ -0,0 +1,63 @@
+        <button type="submit" className="submit-button">
+          ❤️ Send Happy Thought ❤️
+        </button>
+        <div className={inputValue.length > 140 ? "counter red" : "counter"}>
```
I don't get a red counter here. It only gives me -xy characters. The styling for "red" seems to be missing.

El1an3 reviewed (commented):
Good job Cornelia! You can be proud. The app meets all the basic requirements. I like the custom header and that you where able to add the counter. However, the part where the counter should turn red is not working. I think you only styled "counter" but not "red". I also like the scroll to bottom button.

dzc1 reviewed (approved):
Hello Cornelia,

Fantastic job on your project! I'm pleased to inform you that you've successfully met all the criteria required for a passing grade. Your work in the `ThoughtList.jsx` file, particularly with logic stands out impressively, save for one minor hiccup. You might have noticed that when adding a new thought in your app, it necessitates a manual refresh to display. To circumvent this in your `ThoughtList.jsx`, you can simply adjust your useEffect dependency array from:

 ```
/*Fetches thoughts and sets up a fetch interval to update every 60 seconds.*/
  /*Converts to minutes after 60 seconds*/
  useEffect(() => {
    fetchThoughts();
    const fetchInterval = setInterval(fetchThoughts, 60000);
    return () => {
      clearInterval(fetchInterval);
    };
  }, []);
```


to: 

 ```
/*Fetches thoughts and sets up a fetch interval to update every 60 seconds.*/
  /*Converts to minutes after 60 seconds*/
  useEffect(() => {
    fetchThoughts();
    const fetchInterval = setInterval(fetchThoughts, 60000);
    return () => {
      clearInterval(fetchInterval);
    };
  }, [thoughts]);
```
This should do the job !! 


keep up the great work 💪💪💪💪

lunek1 committed:
Update ThoughtList.jsx

lunek1 committed:
Update LikeButton.jsx

Updated with my database

lunek1 committed:
Update ThoughtList.jsx

Updated with my database

lunek1 committed:
Update ThoughtInput.jsx

Updated with my database

lunek1 committed:
Update ThoughtList.jsx

Changed 60+ minutes to convert to h+min

lunek1 committed:
Update ThoughtList.jsx

Added time converter to days, weeks and years

lunek1 committed:
Update ThoughtList.jsx

Updates with the new thought immediately after posting it

lunek1 committed:
Update ThoughtList.jsx

Added sorting button,  old-new posts

lunek1 committed:
Update ThoughtList.css

Added styling for the sort button

lunek1 committed:
Update ThoughtInput.jsx

Fixed the counter, now it doesn't go beyond 140 chars and the text should turn red when it's 30 chars left.

lunek1 committed:
Update ThoughtInput.css

Added red text to the counter once it hits the 30 chars left mark

lunek1 committed:
Update README.md
